gcloud-java-compute/src/main/java/com/google/gcloud/compute/MachineType.java

```diff
@@ -0,0 +1,298 @@
+/*
+ * Copyright 2016 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.gcloud.compute;
+
+import com.google.api.services.compute.model.MachineType.ScratchDisks;
+import com.google.common.base.Function;
+import com.google.common.base.MoreObjects;
+import com.google.common.collect.Lists;
+
+import java.io.Serializable;
+import java.math.BigInteger;
+import java.util.List;
+import java.util.Objects;
+
+public class MachineType implements Serializable {
+
+  static final Function<com.google.api.services.compute.model.MachineType, MachineType>
+      FROM_PB_FUNCTION =
+      new Function<com.google.api.services.compute.model.MachineType, MachineType>() {
+        @Override
+        public MachineType apply(com.google.api.services.compute.model.MachineType pb) {
+          return MachineType.fromPb(pb);
+        }
+      };
+  static final Function<MachineType, com.google.api.services.compute.model.MachineType>
+      TO_PB_FUNCTION =
+      new Function<MachineType, com.google.api.services.compute.model.MachineType>() {
+        @Override
+        public com.google.api.services.compute.model.MachineType apply(MachineType type) {
+          return type.toPb();
+        }
+      };
+
+  private static final long serialVersionUID = -4210962597502860450L;
+
+  private final MachineTypeId machineTypeId;
+  private final Long id;
```

Comment:
Because it says "unsigned" in the [docs](https://cloud.google.com/compute/docs/reference/latest/machineTypes#id), I wonder if there's a chance for overflow?  It seems slightly weird that the api client library would use BigInteger if it could have used a long instead.  In the dns-alpha branch, I think there's a similar situation where I believe we chose to go with BigInteger ([docs](https://cloud.google.com/dns/api/v1/projects#number) and our [code](https://github.com/GoogleCloudPlatform/gcloud-java/blob/dns-alpha/gcloud-java-dns/src/main/java/com/google/gcloud/dns/ProjectInfo.java#L38))